### PR TITLE
more intelligent count

### PR DESCRIPTION
**Pull request by snopoke**
* if query is doing aggregation then count(*) with a subquery
* if no agg is happening then do a simple count(*)

### CODE REVIEW

File: tests/tests.py [hunk 1]
```diff
@@ -257,6 +257,22 @@ def test_count_with_filter(self):
...
+
+        vc.append_column(SimpleColumn('indicator_a'))
+        self.assertEqual(2, vc.count(self.session.connection(), {'username': 'user1'},))
+
+    def test_count_with_nulls(self):
+        vc = QueryContext(
+            "user_table",
+        )
+
+        self.assertEqual(4, vc.count(self.session.connection(),))
```
Comment (sravfeyn) [inline tests/tests.py:274]: `user_table` doesn't have any NULLed columns, do they? That's why I didn't write a test.

Comment (snopoke) [reply tests/tests.py:274]: yes it does - `indicator_{c|d}` are only defined on 2 out of the 4 rows. I ran the test with the old code and it returned `2`

Comment (sravfeyn) [reply tests/tests.py:274]: Cool, makes sense. I thought the query picked username column which is not NULL

**Comment by snopoke** [comment]
released as 0.17.1

**Pull request merged**